server/src/main/java/org/elasticsearch/action/support/nodes/TransportNodesAction.java

```diff
@@ -181,8 +200,9 @@ protected String getTransportNodeAction(DiscoveryNode node) {
         void start() {
             final DiscoveryNode[] nodes = request.concreteNodes();
             if (nodes.length == 0) {
-                // nothing to notify
-                threadPool.generic().execute(() -> listener.onResponse(newResponse(request, responses)));
+                // nothing to notify, so respond immediately, but always fork even if finalExecutor == SAME
+                final String executor = finalExecutor.equals(ThreadPool.Names.SAME) ? ThreadPool.Names.GENERIC : finalExecutor;
+                threadPool.executor(executor).execute(() -> newResponse(task, request, responses, listener));
```

Comment:
Again, not completely related, but means we can assert that we're always on the MANAGEMENT threadpool when computing the response.

Reply:
:+1: